File: server/lib/game_machine/game_systems/remote_echo.rb
module GameMachine
  module GameSystems
    class RemoteEcho < Actor::Base
      include GameMachine::Commands
      aspect %w(EchoTest)

      def post_init(*args)
        chars = [*('a'..'z'),*('0'..'9')].flatten
        str = Array.new(400) {|i| chars.sample}.join
        @test_message = MessageLib::Entity.new.set_id(str)
        #schedule_message('test',10)
      end

      def on_receive(message)
        if message.is_a?(String) && message == 'test'
          return
        end
        GameMachine.logger.info "#{self.class.name} #{message}"
        commands.player.send_message(message,message.player.id)
      end
    end
  end
end

File: server/java/project/src/main/java/com/game_machine/core/RemoteEcho.java
package com.game_machine.core;

import GameMachine.Messages.Entity;
import akka.actor.UntypedActor;

public class RemoteEcho extends UntypedActor {

	public static String name = "GameMachine::GameSystems::RemoteEcho";

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof Entity) {
			Entity entity = (Entity) message;
			PlayerCommands.sendToPlayer(entity, entity.getPlayer().getId());
		}

	}

}
